chat/src/test/java/net/md_5/bungee/api/chat/ComponentsTest.java

```diff
@@ -18,10 +19,24 @@ public static void testDissembleReassemble(BaseComponent[] components)
         Assert.assertEquals( TextComponent.toLegacyText( parsed ), TextComponent.toLegacyText( components ) );
     }
 
-    public static void testDissembleReassemble(String json)
+    public static void testDissembleReassemble(BaseComponent components)
     {
+        String json = ComponentSerializer.toString( components );
         BaseComponent[] parsed = ComponentSerializer.parse( json );
-        Assert.assertEquals( json, ComponentSerializer.toString( parsed ) );
+        Assert.assertEquals( TextComponent.toLegacyText( parsed ), TextComponent.toLegacyText( components ) );
+    }
+
+    public static void testDissembleReassemble(String json, boolean modern)
```

Comment:
both methods shall not be named testDissaembleReassemble, as they are testing different directions

Reply:
This is what they were called already, but yeah, I'd agree with renaming one to the opposite.